dump/dump-to-bucket.py

```diff
@@ -0,0 +1,296 @@
+    if portion is None:
+        # default to an everything slice
+        portion = slice(len(ident_hash_seq))
+    debug(f'Requesting {temperature} {format_} {T.bold}{type_}{T.normal} at {T.yellow}{url}{T.normal}')
+    resp = session.get(url)
+    return io.BytesIO(resp.content), media_type, f'{temperature}-{type_}-json', [ident_hash_seq[portion]]
+
+
+def scrape_version(id, version, host, visited_locs, book=None):
+    """
+    """
+    if book is None:
+        is_book = True
+        type_ = 'book'
+        raw_ident_hash = ident_hash = join_ident_hash(id, version)
+        ident_hash_seq = [(id, version)]
+    else:
+        is_book = False
+        type_ = 'page'
+        raw_ident_hash = join_ident_hash(id, version)
+        ident_hash = ':'.join([join_ident_hash(*book), join_ident_hash(id, None)])
+        ident_hash_seq = [book, (id, version,)]
+
+    base_url = f'https://{host}/contents'
+    base_raw_url = f'{base_url}/{raw_ident_hash}'
+    base_baked_url = f'{base_url}/{ident_hash}'
+    raw_postfix = '?as_collated=0'
+
+    # Request the RAW JSON
+    temperature = 'raw'
+    format_ = 'json'
+    url = f'{base_raw_url}.{format_}{raw_postfix}'
+    # debug(f'Requesting {temperature} JSON {T.bold}{type_}{T.normal} at {T.yellow}{url}{T.normal}')
+    # resp = session.get(url)
+    # yield io.BytesIO(resp.content), 'application/json', f'{temperature}-{type_}-json', [ident_hash_seq[-1]]
+    yield get_content(url, type_, format_, temperature, ident_hash_seq, 'application/json', -1)
+
+    # Save the raw json for later
+    raw_json = resp.json()
+
+    # Request the BAKED JSON
+    temperature = 'baked'
+    format_ = 'json'
+    url = f'{base_baked_url}.{format_}'
+    # debug(f'Requesting {temperature} JSON {T.bold}{type_}{T.normal} at {T.yellow}{url}{T.normal}')
+    # resp = session.get(url)
+    # yield io.BytesIO(resp.content), 'application/json', f'{temperature}-{type_}-json', ident_hash_seq
+    yield get_content(url, type_, format_, temperature, ident_hash_seq, 'application/json', -1)
+
+    # Save the baked json for later
+    baked_json = resp.json()
+
+    # Request the RAW HTML
+    temperature = 'raw'
+    format_ = 'html'
+    url = f'{base_raw_url}.{format_}{raw_postfix}'
+    # debug(f'Requesting {temperature} HTML {T.bold}{type_}{T.normal} at {T.yellow}{url}{T.normal}')
+    # resp = session.get(url)
+    # yield io.BytesIO(resp.content), 'text/html', f'{temperature}-{type_}-html', [ident_hash_seq[-1]]
+    yield get_content(url, type_, format_, temperature, ident_hash_seq, 'text/html', -1)
+
+    # Request the BAKED HTML
+    temperature = 'baked'
+    format_ = 'html'
+    url = f'{base_baked_url}.html'
+    # debug(f'Requesting {temperature} HTML {T.bold}{type_}{T.normal} at {T.yellow}{url}{T.normal}')
+    # resp = session.get(url)
+    # yield io.BytesIO(resp.content), 'text/html', f'{temperature}-{type_}-html', ident_hash_seq
+    yield get_content(url, type_, format_, temperature, ident_hash_seq, 'text/html')
```

Comment:
This is the line that caused the error:

```
Traceback (most recent call last):
  File "./dump-to-bucket.py", line 290, in <module>
    main()
  File "/tmp/rap-spike-lambda/py3/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/rap-spike-lambda/py3/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/tmp/rap-spike-lambda/py3/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/rap-spike-lambda/py3/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "./dump-to-bucket.py", line 286, in main
    dump_in_bucket(scrape(book, host), (bucket, region))
  File "./dump-to-bucket.py", line 264, in dump_in_bucket
    key = gen_filepath(type, ident)
  File "./dump-to-bucket.py", line 62, in gen_filepath
    ids = [join_ident_hash(*i) for i in ident]
  File "./dump-to-bucket.py", line 62, in <listcomp>
    ids = [join_ident_hash(*i) for i in ident]
TypeError: join_ident_hash() missing 1 required positional argument: 'version'
```

It needs to set `portion` to `-1` like the other `get_content` calls before this one.

Reply:
What line? Could you push it so I don't have to decipher what your talking about?

Reply:
n/m figured it out.

Reply:
but then i hit another error

```
File "/Users/mike/Projects/00_openstax/00_content_engineering/rap-spike-lambda/dump/dump-to-bucket.py", line 270, in dump_in_bucket
    key = gen_filepath(type, ident)
  File "/Users/mike/Projects/00_openstax/00_content_engineering/rap-spike-lambda/dump/dump-to-bucket.py", line 78, in gen_filepath
    return func(ids)
  File "/Users/mike/Projects/00_openstax/00_content_engineering/rap-spike-lambda/dump/dump-to-bucket.py", line 73, in <lambda>
    'baked-page-json': lambda i: f'baked/{i[0]}:{i[1]}.json',
IndexError: list index out of range
```